src/read_notifier.rs

```diff
@@ -0,0 +1,65 @@
+use std::{
+    fmt, io,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use async_std::{
+    io::{BufRead, Read},
+    sync::Sender,
+};
```

Comment:
Can we make these single-line item imports? This keeps them consistent with the rest of the project.

```suggestion
use std::fmt;
use std::pin::Pin;
use std::task::{Context, Poll};

use async_std::io::{self, BufRead, Read};
use async_std::sync::Sender;
```

Reply:
Sounds good. I didn't actually write these. Rust analyzer did. I've been using the "replace qualified path with use" code action a lot, which results in this. Will do a final edit pass on them before committing